src/UiPath.Workflow/Activities/ScriptingJitCompiler.cs

```diff
@@ -72,8 +73,13 @@ public override LambdaExpression CompileExpression(ExpressionToCompile expressio
         var results = ScriptingAotCompiler.BuildAssembly(finalCompilation);
         if (results.HasErrors)
         {
+            var errorResults = new TextExpressionCompilerResults
+            {
+                ResultType = results.ResultType,
+            };
+            errorResults.AddMessages(results.CompilerMessages.Where(m => !m.IsWarning));
```

Comment:
After filtering out the warnings you may have nothing left and shouldn't throw the exception.

Reply:
Good point, although if we're inside the `results.HasErrors` if-block, we're already sure that we have at least 1 error message (which is not a warning). So filtering-out the warnings will never result in an empty list :)